src/main/scala/subsystem/CrossingWrapper.scala

```diff
@@ -22,47 +22,48 @@ case class SynchronousCrossing(params: BufferParams = BufferParams.default) exte
 case class RationalCrossing(direction: RationalDirection = FastToSlow) extends SubsystemClockCrossing
 case class AsynchronousCrossing(depth: Int, sync: Int = 3) extends SubsystemClockCrossing
 
-trait HasCrossingMethods extends LazyModule with LazyScope
-{
+class CrossingHelper(parent: LazyModule with LazyScope, arg: SubsystemClockCrossing)(implicit valName: ValName) {
+
   // Detect incorrect crossing connectivity
   private def crossingCheck(out: Boolean, source: BaseNode, sink: BaseNode) {
     InModuleBody {
-      def inside(node: BaseNode) = node.parents.exists(_ eq this)
+      def inside(node: BaseNode) = node.parents.exists(_ eq parent)
       source.inputs.foreach { case (syncSource, _) =>
-        require (inside(syncSource) == out, s"${syncSource.name} must ${if(out)""else"not "}be inside ${name} (wrong .cross direction?)")
+        require (inside(syncSource) == out, s"${syncSource.name} source must ${if(out)""else"not "}be inside ${parent.name} (wrong .cross direction?)")
       }
       sink.outputs.foreach { case (syncSink, _) =>
-        require (inside(syncSink) != out, s"${syncSink.name} must ${if(out)"not "else""}be inside ${name} (wrong .cross direction?)")
+        require (inside(syncSink) != out, s"${syncSink.name} sink must ${if(out)"not "else""}be inside ${parent.name} (wrong .cross direction?)")
       }
     }
   }
 
   // TileLink
 
   def crossTLSyncInOut(out: Boolean)(params: BufferParams = BufferParams.default)(implicit p: Parameters): TLNode = {
-    val sync_xing = this { LazyModule(new TLBuffer(params)).node }
-    crossingCheck(out, sync_xing, sync_xing)
-    sync_xing
+    lazy val sync_xing = LazyModule(new TLBuffer(params))
+    crossingCheck(out, sync_xing.node, sync_xing.node)
+    if (!out) parent { TLNameNode(valName) :*=* sync_xing.node }
+    else      parent { sync_xing.node :*=* TLNameNode(valName) }
   }
 
   def crossTLAsyncInOut(out: Boolean)(depth: Int = 8, sync: Int = 3)(implicit p: Parameters): TLNode = {
     lazy val async_xing_source = LazyModule(new TLAsyncCrossingSource(sync))
     lazy val async_xing_sink = LazyModule(new TLAsyncCrossingSink(depth, sync))
```

Comment:
Do these LazyModules end up taking their name from the CrossingHelper's implicit ValName? Do we want this behaviour?

Reply:
I want the CrossingHelper to determine the name some how. Taking it from the val name was your suggestion originally iirc, haha...  But if we are going to expose the val name of the CrossingHelper as part of e.g. the bus wrapper's API, maybe it would be better to decouple that from the signal name.